packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -124,113 +122,106 @@ export const BootDataProvider = ({
+
+  const logged = initialData?.user as LoggedUser;
+  const shouldRefetch = !!logged?.providers && !!logged?.id;
+  const lastAppliedChangeRef = useRef<Partial<BootCacheData>>();
+
   const {
-    data: bootRemoteData,
+    data: bootData,
     error,
     refetch,
     isFetched,
     dataUpdatedAt,
-  } = useQuery(
+  } = useQuery<Partial<Boot>>(
     BOOT_QUERY_KEY,
     async () => {
       const result = await getBootData(app);
       preloadFeedsRef.current({ feeds: result.feeds, user: result.user });
+      updateLocalBootData(bootData || {}, result);
 
       return result;
     },
     {
-      refetchOnWindowFocus: loggedUser,
+      refetchOnWindowFocus: shouldRefetch,
       staleTime: STALE_TIME,
-      enabled: isExtension ? !!hostGranted : true,
+      enabled: !isExtension || !!hostGranted,
+      placeholderData: initialData,
     },
   );
 
-  useEffect(() => {
-    const boot = getLocalBootData();
-    if (boot) {
-      if (boot?.settings?.theme) {
-        applyTheme(themeModes[boot.settings.theme]);
-      }
+  const loadedFromCache = !!bootData;
+  const { user, settings, alerts, notifications, squads } = bootData || {};
 
-      preloadFeedsRef.current({ feeds: boot.feeds, user: boot.user });
-
-      setCachedBootData(boot);
-    }
-  }, []);
-
-  // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const setBootData = (
-    updatedBootData: Partial<BootCacheData>,
-    update = true,
-  ) => {
-    const cachedData = JSON.parse(storage.getItem(BOOT_LOCAL_KEY));
-    let updatedData = { ...updatedBootData };
-    if (update) {
-      if (lastAppliedChange) {
-        updatedData = { ...lastAppliedChange, ...updatedData };
-      }
-      setLastAppliedChange(updatedData);
-    } else {
-      if (cachedData?.lastModifier !== 'companion' && lastAppliedChange) {
-        updatedData = { ...updatedData, ...lastAppliedChange };
-      }
-      setLastAppliedChange(null);
-    }
-    const updated = updateLocalBootData(cachedData, updatedData);
-    setCachedBootData(updated);
-  };
-
-  useEffect(() => {
-    if (bootRemoteData) {
-      setInitialLoad(initialLoad === null);
-      // We need to remove the settings for anonymous users as they might have changed them already
-      if (!bootRemoteData.user || !('providers' in bootRemoteData.user)) {
-        delete bootRemoteData.settings;
-      }
-
-      setBootData(bootRemoteData, false);
-    }
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [bootRemoteData]);
-
-  useRefreshToken(bootRemoteData?.accessToken, refetch);
+  useRefreshToken(bootData?.accessToken, refetch);
   const updatedAtActive = user ? dataUpdatedAt : null;
+  const updateQueryCache = useCallback(
+    (updatedBootData: Partial<BootCacheData>, update = true) => {
+      const cachedData = JSON.parse(storage.getItem(BOOT_LOCAL_KEY));
```

Comment:
Should we check if this is `null`, eg. storage record does not exist?

Reply:
This never gets `null`, but yeah, strict mode on, we might get that.